internal/consul/consul.go

```diff
@@ -11,6 +11,8 @@ type Client interface {
 	Catalog() Catalog
 	Health() Health
 	KV() KV
+	LockTrento(prefix string) (*consulApi.Lock, error)
+	LockWaitReleasead(prefix string) error
```

Comment:
I'm also unsure about this method name; maybe
```suggestion
	WaitLock(prefix string) error
```

Reply:
Ok, no inconvenience changing to that name